File: BinaryTreeLevelOrderTraversal/first.java
public List<List<Integer>> levelOrder(TreeNode root) {
    List<List<Integer>> result = new ArrayList<List<Integer>>();
    if(root==null) return result;
    ArrayList<TreeNode> level = new ArrayList<TreeNode>();
    level.add(root);

    while(!level.isEmpty()){
        ArrayList<Integer> values = new ArrayList<Integer>();
        for(int i=0; i<level.size(); i++){
            values.add(level.get(i).val);
        }
        result.add(values);
        ArrayList<TreeNode> nextlevel = new ArrayList<TreeNode>();
        for(int j=0 ;j<level.size(); j++){
            if(level.get(j).left!=null) nextlevel.add(level.get(j).left);
            if(level.get(j).right!=null) nextlevel.add(level.get(j).right);
        }
        level = nextlevel;
    }
    return result;
}

File: BinaryTreeLevelOrderTraversal/first.py
class Solution(object):
    def levelOrder(self, root):
        if not root:
            return []

        res = []
        res.append([root.val])


        rootQueue = [root]
        while True:
            childrenQueue = []
            while rootQueue:
                parent = rootQueue[0]
                if parent.left:
                    childrenQueue.append(parent.left)
                if parent.right:
                    childrenQueue.append(parent.right)
                rootQueue.pop(0)
            if childrenQueue:
                res.append([item.val for item in childrenQueue])
                rootQueue = childrenQueue
            else:
                break
        return res
